### PR TITLE
48537 release process

### PR DESCRIPTION
**Pull request by ricellis**
*What*
Provide a release process for java-cloudant

*Why*
There are some bug fixes that need releasing and a simplified and more automated approach to the release will streamline this and future releases.

*How*
* Add gradle for generating release artifacts:
    * client jar
    * source jar
    * javadoc jar
    * maven pom
* Add gradle for signing these artifacts
* Add gradle to upload artifacts to sonatype repositories

*Testing*
Local testing of gradle tasks.

reviewer @rhyshort 
reviewer @mikerhodes

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -87,3 +96,99 @@ jar.dependsOn generateClientPropertiesFile
...
+                    url 'https://java-cloudant@github.com/cloudant/java-cloudant.git'
+                }
+                properties {
+                    'project.build.sourceEncoding' 'UTF-8'
+                }
+                developers {
+                    developer {
+                        id 'IBMCloudant'
+                        name 'IBMCloudant'
+                        email 'github@cloudant.com'
```
Comment (rhyshort) [inline build.gradle]: this should be support@cloudant.com

File: build.gradle [hunk 2]
```diff
@@ -87,3 +96,99 @@ jar.dependsOn generateClientPropertiesFile
...
+                        url 'https://github.com/cloudant/java-cloudant'
+                    }
```
Comment (mikerhodes) [inline build.gradle]: Did you check whether we should just use IBM for this?

File: build.gradle [hunk 3]
```diff
@@ -87,3 +96,99 @@ jar.dependsOn generateClientPropertiesFile
...
+                }
+                properties {
+                    'project.build.sourceEncoding' 'UTF-8'
+                }
+                developers {
+                    developer {
+                        id 'IBMCloudant'
+                        name 'IBMCloudant'
+                        email 'github@cloudant.com'
+                        url 'https://github.com/cloudant/java-cloudant'
```
Comment (rhyshort) [inline build.gradle]: I think this should point to cloudant.com ?

File: build.gradle [hunk 4]
```diff
@@ -87,3 +96,99 @@ jar.dependsOn generateClientPropertiesFile
...
+
+            snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots/") {
+                authentication(userName: ossrhUsername, password: ossrhPassword)
+            }
+
+            //augment the pom with additional information
+            pom.project {
+                packaging 'jar'
+                name project.clientName
+                description 'official Cloudant Client for Java'
```
Comment (rhyshort) [inline build.gradle]: capital `O` on `official`

File: build.gradle [hunk 5]
```diff
@@ -87,3 +96,99 @@ jar.dependsOn generateClientPropertiesFile
...
+                    developerConnection 'scm:git:git@github.com/cloudant/java-cloudant.git'
+                    url 'https://java-cloudant@github.com/cloudant/java-cloudant.git'
+                }
+                properties {
+                    'project.build.sourceEncoding' 'UTF-8'
+                }
+                developers {
+                    developer {
+                        id 'IBMCloudant'
+                        name 'IBMCloudant'
```
Comment (rhyshort) [inline build.gradle]: I think we should have a space here so `IBM Cloudant` if we aren't going to use either `Cloudant` or `IBM` on its own

File: build.gradle [hunk 6]
```diff
@@ -87,3 +96,99 @@ jar.dependsOn generateClientPropertiesFile
...
+                    connection 'scm:git:git://github.com/cloudant/java-cloudant.git'
+                    developerConnection 'scm:git:git@github.com/cloudant/java-cloudant.git'
+                    url 'https://java-cloudant@github.com/cloudant/java-cloudant.git'
+                }
+                properties {
+                    'project.build.sourceEncoding' 'UTF-8'
+                }
+                developers {
+                    developer {
+                        id 'IBMCloudant'
```
Comment (rhyshort) [inline build.gradle]: remove this

File: build.gradle (hunk 3 above)
Comment (rhyshort) [inline build.gradle]: Also make it the organisation URL and put in organisation name

File: build.gradle (hunk 2 above)
Comment (rhyshort) [inline build.gradle]: From looking at [cloudant toolkit]( https://repo1.maven.org/maven2/com/ibm/mobile/services/cloudant-toolkit-local/1.0.0/cloudant-toolkit-local-1.0.0.pom) pom file, MF used Mobile First SDK as the developer, so I think we should use IBM Cloudant, but have IBM as the organisation name with the links pointing to cloudant.com. Although I guess we could have the organisation Url pointing to IBM.com and the developer URL pointing to cloudant.com.

File: build.gradle (hunk 4 above)
Comment (ricellis) [inline build.gradle]: I copy/pasted from the previous pom without checking, I guess "Client" should probably be "client" too

File: build.gradle (hunk 2 above)
Comment (ricellis) [inline build.gradle]: @xpqz agreed with IBM Cloudant, but we will add both e.g.
```groovy
name 'IBM Cloudant'
email 'support@cloudant.com'
url 'https://cloudant.com'
organization 'IBM'
organizationUrl 'http://www.ibm.com'
```

**Comment by ricellis** [comment]
@rhyshort I addressed your comments in 5a7fff9

**Comment by rhyshort** [comment]
+1

**Comment by ricellis** [comment]
ba33efd did not actually set the system properties correctly, 46d0ca5 fixes it

**Comment by mikerhodes** [comment]
+1

**Pull request merged**